### PR TITLE
Align XRPC parameters with subset of json schema

### PR DESCRIPTION
**Pull request by devinivy**
The parameters in an XRPC method were similar to json schema, but not quite on spec. This work aligns them with a subset of json schema that query parameters can reasonably support.

 - Updated lexicons to reflect the updated parameters format.
 - Updated lexicon validation of schemas for json schema subset supported by parameters.
 - Updated client, server codegen to use `json-schema-to-typescript`.
 - Display ts interfaces for parameters in md docs, aligned with input and output.
 - ~Updated md docs generation minimally, kept same logic for documenting params.~
 - ~Show parameter enums in md docs.~
 - Switch to ajv for validation of parameters (allows type coercion, treats `?myparam=` as empty/missing, applies defaults).
 - Apply defaults on all validation (i.e. not just parameters). Client no longer applies parameter defaults, leaving this up to the server.
 - Update validation error messages to specify whether they are for output, input, or parameters.

### CODE REVIEW

**Review by pfrazee — APPROVED** [review]
Looks great, one small change on markdown gen and we're all set

File: packages/lex-cli/src/mdgen/index.ts [hunk 1]
```diff
@@ -91,17 +91,24 @@ async function genMethodSchemaMd(schema: MethodSchema): Promise<StringTree> {
 
   desc.push(`<mark>RPC ${schema.type}</mark> ${schema.description || ''}`, ``)
 
-  if (schema.parameters && Object.keys(schema.parameters).length) {
+  if (schema.parameters && Object.keys(schema.parameters.properties).length) {
```
[the remark this answers is not in the document]
Comment (devinivy) [reply packages/lex-cli/src/mdgen/index.ts]: That sounds good—should be all set now 👍

**Review by dholms — APPROVED** [review]
aww yiss

**Comment by snarfed** [comment]
Thanks for this! I ended up doing the same thing (handling params as subset of JSON Schema) in lexrpc: https://github.com/snarfed/lexrpc/blob/1230ec9178d3c47e4b8471578cc461dc99877e67/lexrpc/base.py#L73-L84

**Pull request merged**